pkg/traceql/ast_validate_test.go

```diff
@@ -46,6 +48,17 @@ func TestExamples(t *testing.T) {
 			require.NoError(t, err)
 			err = p.validate()
 			require.Error(t, err)
+			require.False(t, errors.As(err, &unsupportedError{}))
+		})
+	}
+
+	for _, q := range queries.Unsupported {
+		t.Run("unsupported - "+q, func(t *testing.T) {
+			p, err := Parse(q)
+			require.NoError(t, err)
+			err = p.validate()
+			require.Error(t, err)
```

Comment:
Do we want to make assertions about the type of error being returned here?

Reply:
type is asserted on the next line. i suppose we technically don't need this line

Reply:
Oh indeed yes.  I was thinking require.ErrorAs iirc, but that works too.